pkg/quota/admission/runonceduration/api/types.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// RunOnceDurationConfig is the configuration for the RunOnceDuration plugin.
+// It specifies a default override value for ActiveDeadlineSeconds for a run-once pod.
+// The project that contains the pod may specify a different setting. That setting will
+// take precedence over the one configured for the plugin here.
+type RunOnceDurationConfig struct {
+	unversioned.TypeMeta
+
+	// ActiveDeadlineSecondsOverride is the value to set on containers of run-once pods
+	// Only a positive value is valid. Absence of a value means that the plugin
+	// won't make any changes to the pod
+	ActiveDeadlineSecondsOverride *int64
+
+	// AllowProjectAnnotationOverride indicates whether an override can be set in a project
+	// annotation. The project annotation override will take precedence over the globally configured
+	// override.
+	AllowProjectAnnotationOverride bool
```

Comment:
I'm less concerned about declarative and more about the mental burden of logic.

To me this is simple:
config.Enabled - does this plugin do anything? If so, set a global limit and/or set a project override.

This is more mentally complicated for the end user:

config.AllowProjectAnnotationOverride - set a global limit if you want one, and also set this to true if you want to then be able to set per-project overrides. If you forget to set it, the per-project override you set does nothing but your global limit still applies.